src/Features/Core/Portable/GenerateMember/GenerateConstructor/GenerateConstructorHelpers.cs

```diff
@@ -70,9 +70,9 @@ private static bool ParameterTypesMatch(SemanticDocument document, IList<ITypeSy
                 var type1 = parameterTypes[i];
                 if (type1 != null)
                 {
-                    var type2 = method.Parameters[i].Type;
+                    var type2 = method.Parameters[i]?.Type;
```

Comment:
When do you get an `IMethodSymbol` with null's in it's `Parameters`?  That sounds like a bug in the compiler.